types/wlr_xdg_shell_v6.c

```diff
@@ -922,19 +922,35 @@ static void wlr_xdg_toplevel_v6_send_configure(
 	wl_array_init(&states);
 	if (surface->toplevel_state->pending.maximized) {
 		s = wl_array_add(&states, sizeof(uint32_t));
-		*s = ZXDG_TOPLEVEL_V6_STATE_MAXIMIZED;
+		if (!s) {
+			wlr_log(L_ERROR, "Could not allocate state for maximized xdg_toplevel");
+		} else {
+			*s = ZXDG_TOPLEVEL_V6_STATE_MAXIMIZED;
```

Comment:
Is it possible to abort if no memory is available instead of sending an invalid state to the client?

Reply:
Sure, done.